src/server/accessors/MessageBuilder.ts

```diff
@@ -192,4 +192,17 @@ export class MessageBuilder implements IMessageBuilder {
     public getBlocks() {
         return this.msg.blocks;
     }
+
+    public addCustomField(key: string, value: any): IMessageBuilder {
+        if (!this.msg.customFields) {
+            this.msg.customFields = {};
+        }
+
+        if (this.msg.customFields[key]) {
+            throw new Error(`The message already contains a custom field by the key: ${ key }`);
+        }
+
+        this.msg.customFields[key] = value;
```

Comment:
In the documentation you say it will throw an error when the `key` contains a period but I don't see that check. I know that is something which happens from mongo but it might be better to explicitly declare it here, right?

Reply:
Agreed! Thanks for pointing that, @graywolf336. I've also noticed that this verification is missing in some other classes in the Apps-Engine code.
I may send another pull request later so as to add these other missing exceptions.